tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -754,6 +759,97 @@ static byte TestCoveredIndex1(Span<byte> data)
             return r;
         }
 
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Covered index in-loop redundant bounds check elimination")]
+        public static void CoveredIndex2(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = TestCoveredIndex2(s, 0, length);
+                }
+                return result;
+            },
+            "CoveredIndex2({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestCoveredIndex2(Span<byte> data, int start, int end)
+        {
+            byte x = 0;
+            byte y = 0;
+
+            for (var idx = start; idx < end; idx++)
+            {
+                x ^= data[idx];
+
+                if (idx == 100)
+                {
+                    // Should be able to eliminate this bounds check
+                    y ^= data[0];
+                }
+            }
+
+            byte r = (byte)(y ^ x ^ y);
+
+            return r;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Covered index in-loop redundant bounds check elimination")]
+        public static void CoveredIndex3(int length)
+        {
+            if (length < 50)
+            {
+                throw new Exception("test requires at least 100 byte length");
+            }
+
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = TestCoveredIndex3(s, 0, length);
+                }
+                return result;
+            },
+            "CoveredIndex3({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestCoveredIndex3(Span<byte> data, int start, int end)
+        {
+            byte x = 0;
+            byte y = 0;
+            byte z = 0;
+
+            for (var idx = start; idx < end; idx++)
+            {
+                x ^= data[idx];
+
+                if (idx == 100)
```

Comment:
Agree on the inversion.

Right now all the tests compute the same expected result, which we check.

The `x ^= ...` feeds into the return value so it has to be done every iteration, or else I need to fix my checking scheme to be more flexible.